Added console-window

joeycatxx opened the pull request.

prestonvanloon commented:
What does this do? 
Can you attach a screenshot?
Is there a bug associated with this task?

Thanks

joeycatxx commented:
Added a new component called console-window.
Merged console-window and browser-window into one module - browser.module.

I think one of the folders src/app/browser did not get uploaded. That is why it is failing.

Using GitHub for the first time so I might have missed something.

prestonvanloon commented on src/app/exercise/playground/playground.component.ts:
```diff
@@ -10,6 +10,7 @@ import {SlideComponent} from '../../presentation/slide/slide.component';
   styleUrls: ['./playground.component.css']
 })
 export class PlaygroundComponent extends ExerciseBase {
+  @Input() isConsole: boolean;
```
This is input as false always, does it need to be an `@Input`?

prestonvanloon commented on src/app/exercise/runner/runner.component.html:
```diff
@@ -1,3 +1,6 @@
   <div #runner class="runner"></div>
 </slides-browser-window>
+<slides-console-window *ngIf="isConsole">
+  <div #runnerConsole class="runner"></div>
```
Is this ever shown? It looks like `isConsole` is assigned to false and never reassigned.

prestonvanloon commented on src/app/exercise/runner/runner.component.ts:
```diff
@@ -258,7 +260,9 @@ export class RunnerComponent implements AfterViewInit, OnChanges, OnDestroy {
 
   ngOnChanges(changes: SimpleChanges): void {
-    this.runCode(changes.files.currentValue, this.runnerType);
+    if(this.runnerElement!=null || this.runnerConsoleElement!=null){
```
Isn't this always true? One of those elements should always exist right?

prestonvanloon commented on src/app/exercise/runner/runner.component.ts:
```diff
@@ -336,6 +340,10 @@ export class RunnerComponent implements AfterViewInit, OnChanges, OnDestroy {
   }
 
   ngAfterViewInit() {
+    if(this.runnerElement==null && this.runnerConsoleElement!=null){
```
`if (!this.runnerElement && this.runnerConsoleElement) {`

joeycatxx commented:
Uploaded the missing folder to  the branch. For the typescript​ exercise am passing isconsole as true to get the console window instead of browser window

prestonvanloon commented:
Please run 

- `npm run lint`
- `npm test`

And fix issues.

joeycatxx commented:
Ran lint and fixed errors.
Also executed npm test. Test results here. Havent used this yet so not sure if everything I changed passed or not.
[KarmaTest-1.pdf](https://github.com/AngularNYC/angular-presentation/files/995054/KarmaTest-1.pdf)
[KarmaTest-2.pdf](https://github.com/AngularNYC/angular-presentation/files/995056/KarmaTest-2.pdf)
[KarmaTest-3.pdf](https://github.com/AngularNYC/angular-presentation/files/995055/KarmaTest-3.pdf)

kirjs commented:
Please merge master: `git pull origin master`
